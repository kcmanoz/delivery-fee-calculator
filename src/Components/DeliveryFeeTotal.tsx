interface DeliveryFeeTotalProps {
  deliveryFee: number;
}

const DeliveryFeeTotal: React.FC<DeliveryFeeTotalProps> = ({ deliveryFee }) => {
  return (
    <div className="result-container" data-test-id="fee">
      <h3>Delivery Price: {deliveryFee} â‚¬</h3>
    </div>
  );
};

export default DeliveryFeeTotal;
