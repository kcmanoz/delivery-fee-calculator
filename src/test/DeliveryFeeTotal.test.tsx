import { render, screen } from "@testing-library/react";
import DeliveryFeeTotal from "../Components/DeliveryFeeTotal";

describe("DeliveryFeeTotal", () => {
  it("renders with default props", () => {
    const { container } = render(<DeliveryFeeTotal deliveryFee={0} />);
    expect(container).toMatchSnapshot();
  });

  it("displays the correct delivery fee", () => {
    render(<DeliveryFeeTotal deliveryFee={8} />);
    const deliveryFeeElement = screen.getByTestId("fee");
    expect(deliveryFeeElement.textContent).toBe("Delivery Price: 8 â‚¬");
  });
});
