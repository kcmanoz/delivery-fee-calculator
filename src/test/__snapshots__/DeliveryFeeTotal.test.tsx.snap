// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeliveryFeeTotal renders with default props 1`] = `
<div>
  <div
    class="result-container"
    data-test-id="fee"
  >
    <h3>
      Delivery Price: 
      0
       â‚¬
    </h3>
  </div>
</div>
`;
